holochain/holochain-rust
# code coverage - local + travis + docker

fixes #19 

changes:

- adds dockerignore to speed up builds and avoid copying large folders across
- pin versions of things that have been breaking local/travis recently (remove `+nightly`)
- bump version of rust on travis up to a nightly snapshot
- add code coverage using `cargo-tarpaulin`
- code coverage working on travis
- code coverage on codecov
- docker build for testing
- WIP docker build for alpine instead of ubuntu (installs rust but can't build with cargo properly)

how to test:

- see codecov report in github
- to test locally with docker
  - have docker installed
  - `. docker/build-test`
  - `. docker/run-test`

followups:

- #46
- #45
- #41
- #40
